components/connect/src/dynamo/connect/__init__.py

```
@@ -0,0 +1,1502 @@
+    device: str = "cpu"
+    ptr: int = 0
+    size: int = 0
+
+    def to_descriptor(self) -> Descriptor:
+        """
+        Deserialize the serialized descriptor into a `Descriptor` object.
+        """
+        return Descriptor(data=(self.ptr, self.size, self.device, None))
+
+    @field_validator("device")
+    def validate_memtype(cls, v: str) -> str:
```

Pylint (3.3.7)

[error] Method 'validate_memtype' should have "self" as first argument

(E0213)